tests/functional/TestSmokeTest/tests/block_volume_test.go

```diff
@@ -0,0 +1,73 @@
+// +build functional
+
+//
+// Copyright (c) 2018 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+package functional
+
+import (
+	"testing"
+
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+	"github.com/heketi/tests"
+)
+
+func TestBlockVolumeOperation(t *testing.T) {
+
+	// Setup the VM storage topology
+	defer teardownCluster(t)
+	defer teardownBlock(t)
```

Comment:
I had 2 options to delete block volume as @phlogistonjohn suggested
* integrate with teardown function
* create own block teardown function

all block related test cases go into this file, so I that instead of disturbing the old implementation(which i may introduce new bugs) 
i have created a new function to delete block volumes

Reply:
defer should be before setupCluster.
cleanup functions should get called in defer before calling the other functions if functions call fails, defer will be executed to clean up resources

Reply:
@Madhu-1, normally the pattern that we use is this:

1. call function1
2. immediately afterwards: defer call to cleanup-function-for-function1

The main point here is that there is usually no point in calling the cleanup function for function1 before function1 was called. Also if we follow this pattern, then it is always easy to spot since we group the function and its cleanup function together. It would be ok to reverse the process, to first call defer cleanup_function1 and immediately afterwards call function1, but they are equivalent. 

That said, I don't fully understand @raghavendra-talur 's argument "defer should come after setupCluster, just in case setupCluster fails" -- the order (of function and its defer cleanup) is not that important, it is more important to group together, and we have a habit for the ordering.

But it is important that the cleanup for the later action (in our case block volume stuff) is defer-called later, because the deferred cleanups are called in reverse order.

So I would suggest to
* call `defer teardownCluster()` after `setupCluster()`
* call `defer teardownBlock()` after `setupCluster()` and after `defer teardownCluster()`.

Reply:
@obnoxxx am little confused here, why we should call `defer teardownCluster()` after `setupCluster()`
if `setupCluster()` fails how to defer will get called( which is after `setupCluster`)?